concordium-consensus/benchmarks/leader-election/LeaderElectionBench.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- | A simple benchmark for the leader election implementation.
```

Comment:
Please elaborate a bit on the finer details of the benchmark, so that one that comes to revisit this in the future understands the background, your thoughts and the result.